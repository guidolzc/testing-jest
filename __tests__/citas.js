import Citas from '../js/classes/Citas';

describe('probar la clase de citas ', () =>{

    const citas = new Citas();
    const id = Date.now();
    test('agregar una nueva cita ',()=>{

   
      const citaObj = {
            id,
            mascota: 'hook',
            propietario: 'rafa',
            telefono: '19191919',
            fecha: '10-12-2021',
            hora:'16:00',
            sintomas: 'solo duerme'
        };

        
        citas.agregarCita(citaObj);

        // prueba
        expect(citas).toMatchSnapshot();
    });
    test ('actualizar cita',()=>{
        const citaActualizada = {
            id,
            mascota: 'Nuevo nombre2',
            propietario: 'rafa2',
            telefono: '19191919',
            fecha: '10-12-2021',
            hora:'16:00',
            sintomas: 'solo duerme'
        };
        citas.editarCita(citaActualizada);
        expect(citas).toMatchSnapshot(); // encuentra la funcion actualizada y la va a actualizar al snapshot
    // npm t -- -u para actualizar el snapshot 
    })

    test ('eliminar cita',()=>{
        citas.eliminarCita(id)

        expect(citas).toMatchSnapshot();

    });



});



