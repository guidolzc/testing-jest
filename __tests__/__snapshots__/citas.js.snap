// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`probar la clase de citas  actualizar cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10-12-2021",
      "hora": "16:00",
      "id": 1618606871898,
      "mascota": "Nuevo nombre2",
      "propietario": "rafa2",
      "sintomas": "solo duerme",
      "telefono": "19191919",
    },
  ],
}
`;

exports[`probar la clase de citas  agregar una nueva cita  1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10-12-2021",
      "hora": "16:00",
      "id": 1618606871898,
      "mascota": "hook",
      "propietario": "rafa",
      "sintomas": "solo duerme",
      "telefono": "19191919",
    },
  ],
}
`;

exports[`probar la clase de citas  eliminar cita 1`] = `
Citas {
  "citas": Array [],
}
`;
